mtp_noms_ops/apps/security/views/check.py

```diff
@@ -28,6 +30,44 @@ class CreditsHistoryListView(SecurityView):
     form_class = CreditsHistoryListForm
 
 
+class CheckAssignView(BaseFormView):
+    """
+    Modify assignment of check
+    """
+    form_class = AssignCheckToUserForm
+    id_kwarg_name = 'check_id'
+    object_context_key = 'check'
+    list_url = reverse_lazy('security:check_list')
+
+    def get_success_url(self):
+        return reverse('security:resolve_check', kwargs={'check_id': self.kwargs[self.id_kwarg_name]})
+
+    def get_form_kwargs(self):
+        form_kwargs = super().get_form_kwargs()
+        form_kwargs.update(
+            {
+                'request': self.request,
+                'object_id': self.kwargs[self.id_kwarg_name],
+            },
+        )
+        return form_kwargs
+
+    def get(self, request, *args, **kwargs):
+        return HttpResponseRedirect(self.get_success_url())
+
+    def form_valid(self, form):
+        if self.request.method == 'POST':
```

Comment:
i don't think you need this if statement, this method should only be called by `CheckAssignView.post` which itself is only called when it's a POST request

Reply:
Happy to remove it, it was influenced by the method in the `AcceptCheckView` here: https://github.com/ministryofjustice/money-to-prisoners-noms-ops/commit/afb5af9ecc0e79a59773efd9e8a65408bcb7a46f#diff-ddf134f1183c0cbee2a271878c512447R35-R38

If we remove it here should we remove it in the other `form_valid` method?

Reply:
ah i didn't spot that before. that uses a different base class (`SecurityDetailView`) however. some form views in this app actually submit forms using a GET so have different logic. let's leave the other one for now just in case it's actually necessary there (i haven't checked!).

Reply:
OK, cool I'll pop the other one back in.